frontend/src/main/java/com/google/tripmeout/frontend/storage/PlaceVisitStorage.java

```diff
@@ -0,0 +1,63 @@
+package com.google.tripmeout.frontend.storage;
+
+import com.google.tripmeout.frontend.PlaceVisitModel;
+import java.util.List;
+
+/**
+ * stores PlaceVisit objects
+ */
+public interface PlaceVisitStorage {
+  /**
+   * adds a PlaceVisitModel object to storage
+   * 
+   * @param placeVisit the PlaceVisitModel object to add to storage
+   *
+   * @throws a PlaceAlreadyExists exception if there is a PlaceVisitModel in 
+   *     storage with the same tripId and placeId as placeVisit
+   */
+  public void addPlaceVisit(PlaceVisitModel placeVisit);
```

Comment:
nit: Interfaces don't require public qualifiers for their method definitions. You still need the public identifier on the interface itself, but can remove the rest.